mergin/client.py

```diff
@@ -784,17 +787,19 @@ def add_user_permissions_to_project(self, project_path, usernames, permission_le
         :param usernames: list of usernames to be granted specified permission level
         :param permission_level: string (reader, writer, owner)
         """
-        if permission_level not in ["owner", "reader", "writer"]:
+        if permission_level not in ["owner", "reader", "writer", "editor"]:
             raise ClientError("Unsupported permission level")
 
         project_info = self.project_info(project_path)
         access = project_info.get("access")
         for name in usernames:
             if permission_level == "owner":
                 access.get("ownersnames").append(name)
-            if permission_level == "writer" or permission_level == "owner":
+            if permission_level in ("writer", "owner"):
                 access.get("writersnames").append(name)
-            if permission_level == "writer" or permission_level == "owner" or permission_level == "reader":
+            if permission_level in ("writer", "owner", "editor"):
```

Comment:
For compatibility with older servers, let's add to `editorsnames` only if the server is >= 2024.4

Reply:
I'll check if editorsnames key is in access